src/script/api/script_airport.cpp

```diff
@@ -163,3 +163,10 @@
 
 	return AirportSpec::Get(type)->maintenance_cost;
 }
+
+/* static */ Money ScriptAirport::GetMonthlyMaintenanceCost(AirportType type)
+{
+	if (!IsAirportInformationAvailable(type)) return -1;
+
+	return Money(int(GetMaintenanceCostFactor(type)) * _price[PR_INFRASTRUCTURE_AIRPORT] >> 3);
```

Comment:
no need to convert to int before Money here, i think. Also, C-style cast, so `(Money)(GetMaintenanceCostFactor(..) ...)`

Why the `>> 3` ? (divide by 8?)

Reply:
Hi, the `>> 3` was based on this:
[https://github.com/OpenTTD/OpenTTD/blob/94961732a3667f9b2f38da4be5209b1259db6733/src/station.cpp#L550](url)

`int(GetMaintenanceCostFactor(type)) * _price[PR_INFRASTRUCTURE_AIRPORT]` was because I was getting this error with `GetMaintenanceCostFactor(type) * _price[PR_INFRASTRUCTURE_AIRPORT]`

> Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error (active)		more than one operator "*" matches these operands:	openttd	d:\OpenTTD\OpenTTD-master\src\script\api\script_airport.cpp	171		            built-in operator "arithmetic * arithmetic"
            function template "OverflowSafeInt<T, T_MAX, T_MIN> operator*<T,T_MAX,T_MIN>(int64 a, OverflowSafeInt<T, T_MAX, T_MIN> b)"
            function template "OverflowSafeInt<T, T_MAX, T_MIN> operator*<T,T_MAX,T_MIN>(int a, OverflowSafeInt<T, T_MAX, T_MIN> b)"
            function template "OverflowSafeInt<T, T_MAX, T_MIN> operator*<T,T_MAX,T_MIN>(uint a, OverflowSafeInt<T, T_MAX, T_MIN> b)"
            function template "OverflowSafeInt<T, T_MAX, T_MIN> operator*<T,T_MAX,T_MIN>(byte a, OverflowSafeInt<T, T_MAX, T_MIN> b)"
            operand types are: uint16 * Money


The other question, `GetMaintenanceCostFactor` is not suficient, because I do not know where to get the base price before building the airport. I mean, it is possible to calculate it with [these](https://noai.openttd.org/api/trunk/classAIInfrastructure.html), but that would mean I would have to spend money on building an airport just to compute the maintenance cost. In a money constrained environment, it's more useful to know the maintenance cost without wasting money to retrieve it.


A `GetMonthlyMaintenanceCost` for a road piece, a rail piece , or a canal piece isn't going to be useful, because their costs per piece grows the more pieces there are. Maybe it could be made useful, if we have predicted the exact number of pieces to use on a planned route and calculate the difference between the maintenance cost we have now and the maintenance cost we would have after. It's just that the number of pieces is quite varied between tunnels, bridges, and rails also have signals, ew… complicated.

`return (_price[PR_INFRASTRUCTURE_ROAD] * (roadtype == ROADTYPE_TRAM ? 3 : 2) * num * (1 + IntSqrt(num))) >> 9; // 2 bits fraction for the multiplier and 7 bits scaling.`

`return (_price[PR_INFRASTRUCTURE_RAIL] * GetRailTypeInfo(railtype)->maintenance_multiplier * num * (1 + IntSqrt(total_num))) >> 11; // 4 bits fraction for the multiplier and 7 bits scaling.`

`return (_price[PR_INFRASTRUCTURE_WATER] * num * (1 + IntSqrt(num))) >> 6; // 6 bits scaling.`